lightkurve/correctors/backgroundcorrector.py

```diff
@@ -0,0 +1,139 @@
+        self.tpf = tpf
+        self.aperture_mask = aperture_mask
+        super(BackgroundCorrector, self).__init__(lc=lc)
+
+    def __repr__(self):
+        return 'BackgroundCorrector (LC: {})'.format(self.lc.label)
+
+    def _create_design_matrix(self, background_mask, pixel_components,
+                              spline_n_knots, spline_degree):
+        """Returns a DesignMatrixCollection."""
+        # Estimate the median background over time
+        # Subtract the mean image from each cadence
+        simple_bkg = (self.tpf.flux - np.nanmean(self.tpf.flux, axis=0))
+        # Compute the median background pixel value over time
+        simple_bkg = np.nanmedian(simple_bkg[:, background_mask], axis=1)
+        simple_bkg -= np.percentile(simple_bkg, 5)
+
+        # Background-corrected pixel time series
+        pixels = (self.tpf.flux.transpose([1, 2, 0]) - simple_bkg
+                    ).transpose([2, 0, 1])[:, background_mask]
+
+        dm_pixels = DesignMatrix(pixels, name='pixel_series').pca(pixel_components)
+        dm_bkg = DesignMatrix(simple_bkg, name='background_model')
+        dm_spline = create_spline_matrix(self.lc.time, n_knots=spline_n_knots, degree=spline_degree).append_constant()
+        dm = DesignMatrixCollection([dm_pixels, dm_bkg, dm_spline])
+        return dm
+
+    def correct(self, pixel_components=3, spline_n_knots=100, spline_degree=3,
+                background_mask=None, restore_trend=True, **kwargs):
+        """Returns a systematics-corrected light curve.
+
+        Parameters
+        ----------
+        pixel_components : int
+            Number of principal components derived from the background pixel
+            time series to utilize.
+        background_mask : array-like or None
+            A boolean array flagging the background pixels such that `True` means
+            that the pixel will be used to generate the background systematics model.
+            If `None`, all pixels which are fainter than 1-sigma above the median
+            flux will be used.
+        restore_trend : bool
+            Whether to restore the long term spline trend to the light curve.
+        """
+        if background_mask is None:
+            # Default to pixels <1-sigma above the background
+            background_mask = ~self.tpf.create_threshold_mask(1, reference_pixel=None)
+        self.background_mask = background_mask
+
+        dm = self._create_design_matrix(background_mask=background_mask,
+                                        pixel_components=pixel_components,
+                                        spline_n_knots=spline_n_knots,
+                                        spline_degree=spline_degree)
+        clc = super(BackgroundCorrector, self).correct(dm, **kwargs)
+        if restore_trend:
+            clc += self.diagnostic_lightcurves['spline']
+        return clc
+
+    def diagnose(self):
+        """Returns diagnostic plots to assess the most recent call to `correct()`.
```

Comment:
Is this `diagnose` that much more useful than the vanilla `RegressionCorrector` version?